### ISSUE TITLE
Custom login process

### ISSUE THREAD

**Issue by snothub** [IdentityServer/IdentityServer3#1831]
We have a scenario where we want one legacy app to continue using its own login page but instead of looking up user/password in the database, go over to using token endpoint with the IdSrv. However, if a user is logged in via another app and has a session with the IdSrv, we would like the legacy login page to leverage the SSO and allow the user in. But how can this be done so users that are **not logged in** don't go to authorize endpoint and end up on the IdSrv login page?

Basically, can the user be redirected to authorize endpoint but come back to the return url unrecognized so the legacy login can handle the authentication?

Thanks

**Comment by leastprivilege** [comment]
you can use the prompt=none parameter. This means that you will get back an error if the login page would need to be shown. check the endpoint docs.

**Comment by snothub** [comment]
Ah, of course. Thanks, Dominick.

**Comment by snothub** [comment]
I have to go back to this a bit. If I do the prompt=none for authorize endpoint and the user is not logged in, I'll get an error, that's fine. But, how can I have a custom login page login the user so I get the session with IdSrv for next time user hits authorize endpoint, all without showing IdSrv own login page?

Thanks

**Comment by brockallen** [comment]
If your custom app hosts the login then you will never get SSO at IdSvr.

**Comment by snothub** [comment]
I was expecting that. Isn't that solvable by having the custom app as an external provider and sending idp in the acr values?

**Comment by brockallen** [comment]
This means your app becomes a token service itself. Then why use IdSvr at all?

**Comment by snothub** [comment]
Right, I was hoping it was somehow doable to have the custom app could do the authentication, but leave the token issuing to IdSrv. We can't unplug custom login yet due to legacy issues, so I was looking for a way around this.

**Comment by brockallen** [comment]
You can do the prompt=none trick to easily work when the user first logs into IdSvr, but if they haven't then they'll have to login twice.
